plugin/ansible_vsphere_gosv_log.py

```diff
@@ -930,26 +930,24 @@ def _collect_ansible_gosv_facts(self, result):
                 str(task.action) == "ansible.builtin.set_fact"):
             ansible_facts = task_result.get('ansible_facts', None)
             if ansible_facts:
-                non_empty_facts = dict(filter(lambda item: item[1] != '',
-                                              ansible_facts.items()))
-                self._ansible_gosv_facts.update(non_empty_facts)
-                if ("current_testcase_name" in non_empty_facts and
+                self._ansible_gosv_facts.update(ansible_facts)
+                if ("current_testcase_name" in ansible_facts and
+                    ansible_facts["current_testcase_name"] and
                     self._last_test_id and
                     self._last_test_id in self.test_runs):
                     # Update deploy_vm test case name
-                    self.test_runs[self._last_test_id].name = non_empty_facts['current_testcase_name']
-
-                if (self._ansible_gosv_facts.get('esxi_build','') and
-                    self._ansible_gosv_facts.get('vm_hardware_version','') and
-                    self._ansible_gosv_facts.get('guestinfo_vmtools_info', '')):
-                    # Save guest info
-                    vm_guest_info = VmGuestInfo(self._ansible_gosv_facts)
-                    guestinfo_hash = str(hash("{}{}{}".format(vm_guest_info.ESXi_Build,
-                                                              vm_guest_info.VMTools_Version,
-                                                              vm_guest_info.Hardware_Version)))
-
-                    if guestinfo_hash not in self.collected_guest_info:
-                        self.collected_guest_info[guestinfo_hash] = vm_guest_info
+                    self.test_runs[self._last_test_id].name = ansible_facts['current_testcase_name']
+
+                if task_file == 'vm_get_guest_info.yml':
+                    esxi_build = self._ansible_gosv_facts.get('esxi_build','')
+                    vm_hw_version = self._ansible_gosv_facts.get('vm_hardware_version','')
+                    vmtools_version = self._ansible_gosv_facts.get('guestinfo_vmtools_info','')
+                    if (esxi_build and vm_hw_version and vmtools_version):
+                        guestinfo_hash = str(hash("{}{}{}".format(esxi_build, vm_hw_version, vmtools_version)))
```

Comment:
In the same test run, is this 'esxi_build' the same?

Reply:
Yes, but we need to consider special scenario that user may define a customized playbook or test case to run collect guest info on different ESXi server, with different hardware version and different VMTools.

Reply:
Ok, it's no big deal.